tempodb/blocklist/poller.go

```diff
@@ -145,14 +145,19 @@ func (p *Poller) Do(previous *List) (PerTenant, PerTenantCompacted, error) {
 		return nil, nil, err
 	}
 
-	blocklist := PerTenant{}
-	compactedBlocklist := PerTenantCompacted{}
-
-	consecutiveErrors := 0
+	var (
+		blocklist          = PerTenant{}
+		compactedBlocklist = PerTenantCompacted{}
+		consecutiveErrors  = 0
+	)
 
 	for _, tenantID := range tenants {
 		newBlockList, newCompactedBlockList, err := p.pollTenantAndCreateIndex(ctx, tenantID, previous)
 		if err != nil {
+			if errors.Is(err, backend.ErrDoesNotExist) {
```

Comment:
the goal here is to detect that this tenant had no valid blocks despite being in the tenant list. should we detect that in the child functions and pass a custom error up? instead of keying on DNE?

or maybe a nil err and an empty index which we then ignore?